packages/task/src/browser/task-problem-matcher-registry.ts

```diff
@@ -58,8 +65,11 @@ export class ProblemMatcherRegistry {
             console.error('Only named Problem Matchers can be registered.');
             return Disposable.NULL;
         }
-        const toDispose = new DisposableCollection(Disposable.create(() => {/* mark as not disposed */ }));
-        this.doRegister(matcher, toDispose);
+        const toDispose = new DisposableCollection(Disposable.create(() => {
+            /* mark as not disposed */
```

Comment:
@akosyakov this is the part i am not sure about. 
could you please give me more information on what `/* mark as not disposed */` means here? Thank you !

Reply:
`isDisposed` returns `true` if a collection is empty. If we don't anything on the initialization when checking it when it is empty will return `false`, so we push an empty disposable object to indicate that it was not disposed yet.